### PR TITLE
Add Clamp to the description in gallery & Add Gallery item in Menu

### PR DESCRIPTION
**Pull request by hueitan**
more than 3 lines.

Capture

### CODE REVIEW

**Comment by hueitan** [comment]
This pr includes adding Gallery item into menu.

File: src/components/Gallery.js [hunk 1]
```diff
@@ -12,6 +14,14 @@ const AboutContainer = ({ author, description, license, filePage, close }) => {
     onKeyLeft: () => { window.open(filePage) }
   })
 
+  useLayoutEffect(() => {
+    const descriptionNode = document.querySelector('.description')
```
Comment (stephanebisson) [inline src/components/Gallery.js]: Please don't use global selectors. Put a ref on the top div or the specific element you need.

File: style/gallery.less [hunk 2]
```diff
@@ -62,8 +62,29 @@
 		overflow: hidden;
 
 		&.description {
-			max-height: 3.5em;
-			overflow: hidden;
+			&.clamp {
```
Comment (stephanebisson) [inline style/gallery.less]: Could you look into creating a LESS mixin instead of copying it over?

**Comment by hueitan** [comment]
can you give me the article name of your screenshot example?

File: src/components/Gallery.js (hunk 1 above)
Comment (hueitan) [reply src/components/Gallery.js]: true, I think the above screenshot caused by this global reason. thanks for the catching.

**Comment by hueitan** [comment]
we cannot confirm the ellipsis at the end of the text in 3rd line, we can only insert `...` in the end not the end of the text, see the screenshot

Capture

---

I'm thinking of another solution of having opacity in the bottom text between 3rd and 4th lines

will try and see the possible way to do it then comment on phab.

**Comment by stephanebisson** [comment]
> we cannot confirm the ellipsis at the end of the text in 3rd line, we can only insert `...` in the end not the end of the text

It's true, we have the same problem with clamping on the article title. I think we can let it go into QA and product sign off.

File: src/components/Gallery.js [hunk 3]
```diff
@@ -1,9 +1,12 @@
 import { h } from 'preact'
-import { useState } from 'preact/hooks'
+import { useState, useRef, useLayoutEffect } from 'preact/hooks'
 import { useI18n, useSoftkey, usePopup } from 'hooks'
 
+const MAX_DESCRIPTION_HEIGHT = 44
```
Comment (stephanebisson) [inline src/components/Gallery.js]: How did you arrive at that number? It should be exactly 3 x line-height

Comment (hueitan) [reply src/components/Gallery.js]: I didn't set the line-height for the text here, and 44 is the best match with `...`

or I can set the `line-height: 15px` and `max-height: 45px`

**Pull request merged**